src/raftstore/store/peer.rs

```diff
@@ -459,13 +462,40 @@ impl Peer {
             ready.hs.take();
         }
 
+        if let Some(ref soft_state) = ready.ss {
+            if soft_state.leader_id == raft::INVALID_ID {
+                if self.leader_missing_time.is_none() {
+                    self.leader_missing_time = Some(Instant::now())
+                }
+            } else if self.is_initialized() {
```

Comment:
So we can use `is_initialized() && has_leader` to clear the leader_missing_time. 
For the un-initialized peer, if it doesn't receive any message from leader for a long time, we can think it is stale. 
I suggest checking this in tick too.